### DISCUSSION TITLE
Standalone Java files support

### DISCUSSION THREAD

**Discussion by ThomasMConnors** [redhat-developer/vscode-java#2384] [Q&A]
VSCode extension says it supports "Standalone Java files support"

I'm a teacher with a lot of simple .java files to grade. The students have not and will not be using packages as this is a rather simple class in one file.

Having one .java file runs fine. When a second one or more are added, VSCode compiles all the files in the folder. Needless to say, some student files may contain syntax errors. So I can't run or debug the one student file I'm grading. An ability to debug each file is important to give the best possible feedback on the student submission.

Is there a workaround for this situation? It seems to be an option under Java specification.

**7.4.2. Unnamed Packages** states:

An implementation of the Java SE Platform must support at least one unnamed package. An implementation may support more than one unnamed package, but is not required to do so. Which ordinary compilation units are in each unnamed package is determined by the host system.

**Comment by fbricon** [comment]
How are your files organized? 1 folder per student, same class/file name for everyone? (eg. student1/Ex1.java, student2/Ex1.java, ...) or 1 exercise folder, each student bring their own file (eg. exercise1/Student1.java, exercise1/Student2.java ...)?
Both structures work fine in vscode.
In the former case, you would open, student1 folder in vscode, then call `File > Add tfolder to workspace...`, select all other folders at once). In the latter case, just open the exercise1/ folder. 
In all cases you can run each class individually with the vscode-java-debug extension installed, regardless of side-effects caused by compilation errors in other classes.

<img width="1613" alt="Screenshot 2022-03-30 at 23 16 37" src="https://user-images.githubusercontent.com/148698/160932346-8cb2f8e7-6aac-4cf8-a760-aa8f2d7008e4.png">

**Comment by fbricon** [comment]
As for "multiple Unnamed Packages" support, I'm not aware this is supported by the [Eclipse Compiler for Java](https://wiki.eclipse.org/JDT_Core_Programmer_Guide/ECJ).
